kasa/cli/discover.py

```diff
@@ -68,6 +66,60 @@ async def print_discovered(dev: Device):
                 discovered[dev.host] = dev.internal_state
             echo()
 
+    await _discover(ctx, print_discovered, print_unsupported)
+
+    echo(f"Found {len(discovered)} devices")
+    if unsupported:
+        echo(f"Found {len(unsupported)} unsupported devices")
+    if auth_failed:
+        echo(f"Found {len(auth_failed)} devices that failed to authenticate")
+
+    return discovered
+
+
+@discover.command()
+@click.pass_context
+async def list(ctx):
+    """List devices in the network in a table using udp broadcasts."""
+    sem = asyncio.Semaphore()
+    discovered = dict()
+
+    async def print_discovered(dev: Device):
+        cparams = dev.config.connection_type
+        infostr = (
+            f"{dev.host:<15} {cparams.device_family.value:<20} "
+            f"{cparams.encryption_type.value:<4}"
+        )
+        async with sem:
+            try:
+                await dev.update()
+            except AuthenticationError:
+                echo(f"{infostr} - Authentication failed")
+            else:
+                discovered[dev.host] = dev.internal_state
+                echo(f"{infostr} {dev.alias}")
+
+    async def print_unsupported(unsupported_exception: UnsupportedDeviceError):
+        if res := unsupported_exception.discovery_result:
+            echo(f"{res.get('ip'):<15} Unsupported device")
+
+    echo(f"{'HOST':<15} {'DEVICE FAMILY':<20} {'ENCRYPTION TYPE':<4} {'ALIAS'}")
```

Comment:
This looks really cool and useful now that I tested it out! Unsure how dependent we want the cli to be from rich, but it would offer plenty of nice features, including support for tables: https://rich.readthedocs.io/en/stable/tables.html

Reply:
Done in https://github.com/python-kasa/python-kasa/pull/1183